src/Entity/City.php

```diff
@@ -19,6 +21,13 @@ class City
      */
     private $events;
 
+    /**
+     * Many Cities have one Academy. This is the owning side.
+     * @ManyToOne(targetEntity="Academy", inversedBy="cities")
+     * @JoinColumn(name="academy_id", referencedColumnName="id")
+     */
+    private $academy;
```

Comment:
City Entity must not be aware of any other relations in which city is used. City is an standalone entity which may or may not exist without academy or other relations to it.

Reply:
Shall I remove it?

Reply:
If you have a really good reason why City must know about academies, probably you can keep it.
Otherwise, city shouldn't know about where City is used (is it used in programs, academies, or w/e else)

Reply:
I am open to suggestions, trying to learn :)

Reply:
Probably the best solution would be to give json which represents the structure of database. something like this:
`{
  "name": "academy_name",
  "email": "academy@email.com",
  "programs": [
    {
      "name": "program1",
      "startDate": "2019-09-01",
      "endDate": "2020-02-01",
      "starRating": 4.6,
      "city": {
        "address": "some adress",
        "zip Code": "88654",
        "country": "netru"
      }
    }
    ]
}`

Reply:
Shouldn't we do many to many relation here? Something like https://symfonycasts.com/screencast/doctrine-relations/many-to-many